### PR TITLE
ast_utils: Added support for WithMeta (Initial)

### PR DESCRIPTION
**Pull request by erezsh**

### CODE REVIEW

File: lark/ast_utils.py [hunk 1]
```diff
@@ -19,15 +19,17 @@ class AsList(object):
...
     """
 
-def camel_to_snake(name):
-    return re.sub(r'(?<!^)(?=[A-Z])', '_', name).lower()
+class WithMeta(object):
+    """Abstract class
 
-def _call(func, _data, children, _meta):
-    return func(*children)
+    Subclasses will be instanciated the Meta instance of the tree. (see ``v_args`` for more detail)
```
Comment (MegaIng) [inline lark/ast_utils.py:25]: This is not a correct sentence I think.

Comment (MegaIng) [reply lark/ast_utils.py:25]: Also, at what position will it be provided?

**Review by MegaIng — CHANGES_REQUESTED** [review]

File: lark/ast_utils.py (hunk 1 above)
Comment (erezsh) [reply lark/ast_utils.py:25]: The same position `v_args` does. Which means first argument, but second if using `AsList`.

And yes, it's not ideal. I plan to switch them around for v1.0 so meta is always first.

Comment (erezsh) [reply lark/ast_utils.py:25]: Why isn't it correct?

Comment (MegaIng) [reply lark/ast_utils.py:25]: I think there is at least a `with` missing

Comment (erezsh) [reply lark/ast_utils.py:25]: Right, I missed that :)

Thanks!

**Review by MegaIng — APPROVED** [review]

**Pull request merged**